models/scripts/SentimentAnalysis.py

```diff
@@ -0,0 +1,69 @@
+from tabpy_tools.client import Client
+import pandas as pd
+from textblob import TextBlob
+from nltk.sentiment import SentimentIntensityAnalyzer
+import sys
+import configparser
+import getpass
+from pathlib import Path
+
+
+def SentimentAnalysis(_arg1, library='nltk'):
+    '''
+    Sentiment Analysis is a procedure that assigns a score from -1 to 1
+    for a piece of text with -1 being negative and 1 being positive. For
+    more information on the function and how to use it please refer to
+    tabpy-tools.md
+    '''
+    assert (type(_arg1[0]) is str)
+
+    library = library.lower()
+    supportedLibraries = {'nltk', 'textblob'}
+
+    assert (library in supportedLibraries)
+    scores = []
+    if library == 'nltk':
+        sid = SentimentIntensityAnalyzer()
+        for text in _arg1:
+            sentimentResults = sid.polarity_scores(text)
+            score = sentimentResults['compound']
+            scores.append(score)
+    elif libary == 'textblob':
+        for text in _arg1:
+            currScore = TextBlob(text)
+            scores.append(currScore.sentiment.polarity)
+    return scores
+
+
+if __name__ == '__main__':
+    #running from setup.py
+    if (len(sys.argv) > 1):
```

Comment:
Lines 40-64 are duplicated. This is a common functionality which should live in its own function and be reused by all models.

Reply:
fixed